parser/src/parser.cpp

```diff
@@ -176,9 +179,9 @@ void incrementalList(cc::parser::ParserContext& ctx_)
  */
 void incrementalCleanup(cc::parser::ParserContext& ctx_)
 {
-  cc::util::OdbTransaction {ctx_.db} ([&]
+  for (auto& item : ctx_.fileStatus)
```

Comment:
While for maintaining the C++ parser database, the rollback log became enormous when handling the whole operation in a single transaction; one may argue this will not be the issue when only the `File`, `FileContent`, `BuildAction`, `BuildLog`, `BuildSource` and `BuildTarget` tables are considered. Yet it seems a more scalable solution.
Shall we do a performance test on it?